core/mutator/mutator.go

```diff
@@ -45,3 +51,11 @@ type Mutator interface {
 	// Mutate applies mutation to value
 	Mutate(value, mutation []byte) ([]byte, error)
 }
+
+// Mutation reads and writes mutations to the database.
+type Mutation interface {
+	// Read reads all mutations for a specific given mapID, epoch, and index.
+	Read(ctx context.Context, txn transaction.Txn, epoch int64, index []byte) ([][]byte, error)
```

Comment:
For now, there's no ordering, this will be addressed in #487.
However, it's a good idea to modify the database to only store a single mutation per epoch per index.